### PR TITLE
Use conversion context instead of options for interior functions

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: src/ConversionContext.h [hunk 1]
```diff
@@ -16,7 +16,16 @@ namespace drafter {
         refract::Registry registry;
 
     public:
+        bool generateSourceMap;
```
Comment (klokane) [inline src/ConversionContext.h]: why not directly `const WrapperOptions&`?
Will we add duplicit member fer every change in options?

Comment (pksunkara) [inline src/ConversionContext.h]: I don't think `WrapperOptions` will survive our public API cleaning. So, we can change this when that happens.

Comment (klokane) [inline src/ConversionContext.h]: uh, when I argue by "future" it is wrong and have to remove code :)

but there is another argue - pollution of class interface with every one new member.
It should be as clear as possible.

Related things should grouped.

**Comment by w-vi** [comment]
I think the options and context are pretty close to each other and the public API change will resolve them, that's why I merged it without comment

File: src/ConversionContext.h (hunk 1 above)
Comment (pksunkara) [inline src/ConversionContext.h]: I understand and I thought of actually doing the same thing, but I never liked the `WrapperOptions` and I am pretty sure we will remove it soon.

**Comment by pksunkara** [comment]
Basically, we will merging `WrapperOptions` and `snowcrash::BlueprintOptions` into a single thing and we should use that directly in `Context` when we clean the API

**Comment by klokane** [comment]
At least this members should be `const`...

**Comment by pksunkara** [comment]
I think I will make the PR. That way, we will remember to change this later.

**Comment by w-vi** [comment]
`const`ness yes I overlooked that. good point @klokane

**Comment by pksunkara** [comment]
https://github.com/apiaryio/drafter/pull/297

**Pull request merged**